### PR TITLE
Add `dim` arg to `softmax` to avoid warnings

### PR DESCRIPTION
**Pull request by Balandat**
Before this change, the code would emit the following warning:
```
UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument.
```

### CODE REVIEW

File: botorch/optim/initializers.py [hunk 1]
```diff
@@ -670,7 +670,7 @@ def gen_one_shot_hvkg_initial_conditions(
     # sampling from the optimizers
     eta = options.get("eta", 2.0)
     if num_optim_restarts > 0:
-        probs = torch.nn.functional.softmax(eta * standardize(fantasy_vals))
+        probs = torch.nn.functional.softmax(eta * standardize(fantasy_vals), dim=-1)
```
[the remark this answers is not in the document]
Comment (Balandat) [reply botorch/optim/initializers.py:673]: `-1` is just the default behavior when you call this on a multi-dimensional tensor, so I went with that. The alternative is `0` but we do need to pass something to avoid the warning.

[the remark this answers is not in the document]
Comment (esantorella) [reply botorch/optim/initializers.py:673]: The default is `dim=0` for tensors that are 0d, 1d, or 3d and otherwise 1, for some reason. https://github.com/pytorch/pytorch/blob/7d5e8c1d5131d3dfbcd47bb21748563346929b03/torch/nn/functional.py#L1790

[the remark this answers is not in the document]
Comment (Balandat) [reply botorch/optim/initializers.py:673]: Weird. Let me use 0 then to be consistent with that (whatever that means).
